test/integration/resourcemanager/garbagecollector/garbagecollector_test.go

```diff
@@ -0,0 +1,219 @@
+					},
+				},
+				Spec: appsv1.StatefulSetSpec{
+					Template: podTemplateSpec(corev1.RestartPolicyAlways),
+					Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}},
+				},
+			},
+
+			&appsv1.DaemonSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: testNamespace.Name,
+					Annotations: map[string]string{
+						"reference.resources.gardener.cloud/secret-foo":    "secret2",
+						"reference.resources.gardener.cloud/configmap-foo": "configmap4",
+					},
+				},
+				Spec: appsv1.DaemonSetSpec{
+					Template: podTemplateSpec(corev1.RestartPolicyAlways),
+					Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}},
+				},
+			},
+
+			&batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: testNamespace.Name,
+					Annotations: map[string]string{
+						"reference.resources.gardener.cloud/secret-foo":    "secret3",
+						"reference.resources.gardener.cloud/configmap-foo": "configmap3",
+					},
+				},
+				Spec: batchv1.JobSpec{
+					Template: podTemplateSpec(corev1.RestartPolicyNever),
+				},
+			},
+
+			&batchv1beta1.CronJob{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: testNamespace.Name,
+					Annotations: map[string]string{
+						"reference.resources.gardener.cloud/secret-foo":    "secret4",
+						"reference.resources.gardener.cloud/configmap-foo": "configmap2",
+					},
+				},
+				Spec: batchv1beta1.CronJobSpec{
+					Schedule: "5 4 * * *",
+					JobTemplate: batchv1beta1.JobTemplateSpec{
+						Spec: batchv1.JobSpec{
+							Template: podTemplateSpec(corev1.RestartPolicyNever),
+						},
+					},
+				},
+			},
+
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: testNamespace.Name,
+					Annotations: map[string]string{
+						"reference.resources.gardener.cloud/secret-foo":    "secret5",
+						"reference.resources.gardener.cloud/configmap-foo": "configmap1",
+					},
+				},
+				Spec: podTemplateSpec(corev1.RestartPolicyAlways).Spec,
+			},
+		}
+
+		for _, obj := range referencingResources {
+			Expect(testClient.Create(ctx, obj)).To(Succeed())
+		}
+
+		for _, obj := range garbageCollectableObjects {
+			Expect(testClient.Create(ctx, obj)).To(Succeed())
+		}
+
+		Eventually(func(g Gomega) []corev1.Secret {
+			secretList := &corev1.SecretList{}
+			g.Expect(testClient.List(ctx, secretList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return secretList.Items
+		}).Should(onlyContainObject(withName("secret6")))
+
+		Eventually(func(g Gomega) []corev1.ConfigMap {
+			configMapList := &corev1.ConfigMapList{}
+			g.Expect(testClient.List(ctx, configMapList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return configMapList.Items
+		}).Should(onlyContainObject(withName("configmap0")))
```

Comment:
I think here you rather want to assert that this ConfigMap will be garbage collected and will be no longer present in the list. Right now the assertion verifies that the Configmap is present in the list. And I think this currently contradicts with the test goal (feel free to correct me if I miss something).

Probably something like?
```suggestion
		}).Should(Not(ContainElement(withName("configmap0"))))
```

Reply:
Yes, indeed, nice catch, thank you very much. I made it more clear now based on your suggestion!